src/tasks/WorkloadBuildTasks/PatchNuGetConfig.cs

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+#nullable enable
+
+namespace Microsoft.Workload.Build.Tasks;
+
+/*
+ * Used for patching a nuget.config to:
+ *
+ * 1. Add a new package source to the nuget.config
+ * 2. Add a new package source mapping to the nuget.config
+ *
+ * This is useful specifically the case of workload testing
+ */
+public class PatchNuGetConfig : Task
+{
+    [Required, NotNull]
+    public string? TemplateNuGetConfigPath { get; set; }
+
+    [Required, NotNull]
+    public string?        LocalNuGetsPath    { get; set; }
+
+    public string? OutputPath { get; set; }
+
+    /*
+     * Value: ["*Aspire*", "Foo*"]
+     * This will be translated to:
+     * <packageSourceMapping>
+     *  <packageSource key="nuget-local">
+     *    <package pattern="*Aspire*" />
+     *    <package pattern="Foo*" />
+     *  </packageSource>
+     *
+     * This is useful when using Central Package Management (https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management)
+    */
+    public string[] NuGetConfigPackageSourceMappings { get; set; } = Array.Empty<string>();
+
+    public string   PackageSourceNameForBuiltPackages { get; set; } = "nuget-local";
+
+    public override bool Execute()
+    {
+        try
+        {
+            Validate(TemplateNuGetConfigPath, PackageSourceNameForBuiltPackages, OutputPath);
+            GetNuGetConfig(TemplateNuGetConfigPath, LocalNuGetsPath, PackageSourceNameForBuiltPackages, NuGetConfigPackageSourceMappings, OutputPath!);
+            Log.LogMessage(MessageImportance.Low, $"Generated patched nuget.config at {OutputPath}");
+            return true;
+        }
+        catch (LogAsErrorException laee)
+        {
+            Log.LogError(laee.Message);
+            return false;
+        }
+    }
+
+    private static void Validate(string? templateNuGetConfigPath, string? packageSourceNameForBuiltPackages, string? outputPath)
+    {
+        if (string.IsNullOrEmpty(templateNuGetConfigPath))
+            throw new LogAsErrorException($"{nameof(templateNuGetConfigPath)} is required");
```

Comment:
In the doc we say `$(TemplateNuGetConfigPathForWorkloadTesting) - optional`. But here it's required. Am I missing something?

Reply:
Excellent catch!

Reply:
fixed.